### ISSUE TITLE
pylint doesn't complain about undefined name in string type annotation 

### ISSUE THREAD

**Issue by lggruspe** [PyCQA/pylint#7394]
### Bug description

example.py:

```python
# pylint: disable=missing-docstring

example1: undefined
# 😄 pylint complains

example2: "undefined"
# 😢 pylint doesn't
```

### Command used

```shell
pylint example.py
```

### Pylint output

```shell
************* Module example
example.py:3:10: E0602: Undefined variable 'undefined' (undefined-variable)

-----------------------------------
Your code has been rated at 0.00/10
```

### Expected behavior

I think pylint should complain in both cases.

### Pylint version

```shell
pylint 2.15.0
astroid 2.12.5
Python 3.10.6 (main, Aug  2 2022, 00:00:00) [GCC 12.1.1 20220507 (Red Hat 12.1.1-1)]
```

**Comment by Pierre-Sassoulas** [comment]
The typical use case for this is "undefined" being defined in a ``if TYPE_CHECKING:`` guard ie.:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    import defined_conditionally

example1: defined_conditionally
# 😄 pylint complains

example2: "defined_conditionally"
# 😢 pylint doesn't
```
 But pylint does not understand control flow well right now so we avoided a false positive at the cost of a false negative here. If we handle control flow better in astroid in the future then we could do this without the false positive. (I would advice checking string type annotation with mypy or pyright that are type checker).

**Comment by cdce8p** [comment]
Quoted annotations are just strings at runtime and won't break anything. This will run just fine.
```py
example2: "undefined"
```

In contrast here `undefined` is an unresolved reference and Python would raise a `NameError`.
```py
example1: undefined
```

As @Pierre-Sassoulas already suggested, to validate annotations, type checkers (like mypy or pyright) are much better and both will issue warnings for it.

I suggest we close this issue.

**Comment by Pierre-Sassoulas** [comment]
This this could be fixed if #4795. Closing for the reason Marc listed.
